### ISSUE TITLE
'documents.body' type in 'Pages.Platform.init' isn't usable

### ISSUE THREAD

**Issue by tgelu** [dillonkearns/elm-pages#115]
The type of `Pages.Platform.init` has the following type definition for the `documents` property:

```elm
...
    , documents :
          List
              { extension : String
              , metadata : Decoder metadata
              , body : String -> Result String view
              }
...
```

If my `documents.body` function returns `Err String` because the markdown document has syntax errors then that's what I end up getting in the generated `html` file.

This is unexpected and not very useful. It would be nice if this `body` would fail the build in case the return value is `Err _` but probably in such case it would be more suitable that the `body` would be a `Decoder` of some sort, though I am just speculating.

**Comment by dillonkearns** [comment]
Hello @tgelu, thank you for reporting the issue. This is a regression, and it's definitely not the intended behavior. I'll try to get a fix out for this soon. Sorry for the inconvenience, this is definitely an important one!

**Comment by dillonkearns** [comment]
I found some related Webpack issues:

https://github.com/webpack/webpack/issues/708#issuecomment-637743450

https://github.com/bigcommerce/cornerstone/pull/1457

I'm not exactly sure what the ideal solution is, but I think I found a line where I can just check if there are any errors upon completion and do `process.exit(1)`. The output is a little messy, but at least it will fail the build as expected. So I'll go with that solution for now.

Any ideas for a cleaner way to do it are very much welcome!
